feat(template): Adding flit as a build-system option (#136)
osl-incubator/scicookie: src/scicookie/{{cookiecutter.project_slug}}/docs/contributing.md

For my information, check the official documentation:
https://packaging.python.org/en/latest/discussions/src-layout-vs-flat-layout/

{% if cookiecutter.build_system == "poetry" %}

> In addition, you should know that to compile the system of our package we use

Reviewer: ```suggestion
In addition, you should know that to build our package we use
```

[Poetry](https://python-poetry.org/), it's a Python package management tool that
simplifies the process of building and publishing Python packages. It allows us
to easily manage dependencies, virtual environments and package versions. Poetry
also includes features such as dependency resolution, lock files and publishing
to PyPI. Overall, Poetry streamlines the process of managing Python packages,
making it easier for us to create and share our code with others.

{%- elif cookiecutter.build_system == "flit" %}

> In addition, you should know that to compile the system of our package we use

Reviewer: ```suggestion
In addition, you should know that to build our package we use
```
